crates/nu-protocol/src/process/child.rs

```diff
@@ -23,7 +23,21 @@ impl ExitStatusFuture {
             ExitStatusFuture::Finished(Err(err)) => Err(err.as_ref().clone()),
             ExitStatusFuture::Running(receiver) => {
                 let code = match receiver.recv() {
-                    Ok(Ok(status)) => Ok(status),
+                    Ok(Ok(status)) => {
+                        #[cfg(unix)]
+                        if let ExitStatus::Signaled {
+                            signal,
+                            core_dumped: true,
+                        } = status
+                        {
+                            return Err(ShellError::CoredumpErrorSpanned {
+                                msg: format!("program coredumped, receive signal: {signal}"),
```

Comment:
Just a tweak to the grammar.
```suggestion
                                msg: format!("coredump detected. receive signal: {signal}"),
```